core/iwasm/aot/aot_runtime.c

```diff
@@ -1381,7 +1381,17 @@ aot_module_malloc(AOTModuleInstance *module_inst, uint32 size,
     }
 
     if (!addr) {
-        aot_set_exception(module_inst, "out of memory");
+        if (memory_inst->heap_handle.ptr
+            && mem_allocator_is_heap_corrupted(memory_inst->heap_handle.ptr)) {
+            LOG_ERROR("Error: app heap is corrupted, if the wasm file "
+                      "is compiled by wasi-sdk-12.0 or larger version, "
+                      "please add -Wl,--export=malloc -Wl,--export=free "
+                      " to export malloc and free functions.");
```

Comment:
why do you recommend to export malloc/free?

do you just mean it can be a workaround to avoid app heap corruption? (because these export disables app heap)

or, is there any incompatibility between wasi-sdk-12 and app heap, which can cause app heap corruptions?